install/kubernetes/cilium/templates/hubble-relay/deployment.yaml

```diff
@@ -164,3 +164,14 @@ spec:
           {{- end }}
       {{- end }}
 {{- end }}
+
+{{- define "hubble-relay.probecmd" -}}
+command:
+- grpc_health_probe
+- -addr=localhost:{{ .Values.hubble.relay.listenPort }}
+{{- if and .Values.hubble.tls.enabled  .Values.hubble.relay.tls.server.enabled }}
+- -tls
+# We don't know the server name in every case. Not verifying the cert for health checks should be fine
```

Comment:
We should also check for `Values.hubble.relay.tls.server.mtls` - in that case the probe probably won't work at all, because there is no client cert for the probe.

It might be overkill, but could we expose the health service in a unprotected localhost socket to avoid issues with TLS?

Reply:
Having a localhost socket would also allow us to use the [built-in Kubernetes health checking](https://kubernetes.io/blog/2018/10/01/health-checking-grpc-servers-on-kubernetes/) in the future, so we would kill two birds with one stone. But it all depends on how much more complex it would make the code I think.

Reply:
Ah right, we'd completely avoid all the TLS issues for built-in health checks. That convinced me. I'll see if I can expose the health service on a local socket.

Reply:
Turns out only listening on localhost doesn't work as the health probe doesn't come from localhost (duh..). But I don't think exposing the health server unencrypted is a security risk, so I added another gRPC server that always runs unencrypted. That means for newer kubernetes versions we can always use the built-in gRPC health probes.

Reply:
> Turns out only listening on localhost doesn't work as the health probe doesn't come from localhost (duh..).

That's interesting. We are using a localhost *HTTP* health probe both for agent and operator. There, you can configure the incoming interface via the `host` field: https://github.com/cilium/cilium/blob/33573bd0e7e797e25d81ab183e742cb348729d95/install/kubernetes/cilium/templates/cilium-agent/daemonset.yaml#L141

I wonder why the gRPC probe (at least according to the docs) does not support that :confused: